tests/ui/inference/need_type_info/issue-109905.rs

```diff
@@ -0,0 +1,24 @@
+// Test that we show the correct type parameter that couldn't be inferred and that we don't
+// end up stating nonsense like "type parameter `'a`" which we used to do.
+
+// revisions: qualified unqualified
+
+trait Trait<'a, T> {
+    fn m(self);
+}
+
+impl<'a, A> Trait<'a, A> for () {
+    fn m(self) {}
+}
+
+fn main() {
+    #[cfg(qualified)]
+    <() as Trait<'static, _>>::m(());
+    //[qualified]~^ ERROR type annotations needed
+    //[qualified]~| cannot infer type of the type parameter `T`
+
+    #[cfg(unqualified)]
```

Comment:
Why does this test need revisions and cfgs?
I think it can report both errors at the same time, without cfgs.

Reply:
Only one of two errors is shown when I remove the `cfg`s. See for yourself: [playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=9ffa0073cb6a0b0cd3dac37867b8ae76).  They seem to be resolved in different passes.

`cfg` is probably overkill for this test, I'm gonna put the two paths into two different functions which also fixes the problem.